lab_3_generate_by_ngrams/main.py

```diff
@@ -270,6 +415,33 @@ def run(self, seq_len: int, prompt: str) -> Optional[str]:
         In case of corrupt input arguments or methods used return None,
         None is returned
         """
+        if not isinstance(seq_len, int) or not isinstance(prompt, str) or len(prompt) == 0:
+            return None
+
+        encoded_text_tuple = self._text_processor.encode(prompt)
+        if encoded_text_tuple is None:
+            return None
+        # context_len = self._model.get_n_gram_size() - 1
+        encoded_text_list = list(encoded_text_tuple)
+
+        for i in range(seq_len):
+            possible_tokens = self._model.generate_next_token(encoded_text_tuple)
+            if possible_tokens is None:
+                return prompt + '.'
+            if len(possible_tokens) == 0:
+                break
+            # max_value = sorted(list(possible_tokens.values()), reverse=True)[0]
+            # max_freq_tokens = [(key, value) for key, value in possible_tokens.items() if value == max_value]
+            possible_tokens_list = list(possible_tokens.items())
+            possible_tokens_list.sort(key=lambda x: x[0], reverse=True)
+            possible_tokens_list.sort(key=lambda x: x[1], reverse=True)
+
+            encoded_text_list.append(possible_tokens_list[0][0])
+            encoded_text_tuple = tuple(encoded_text_list)
+
+        decoded_text = self._text_processor.decode(encoded_text_tuple) + '.'
```

Comment:
Dot at the end should be handled by post-processing

Reply:
fixed